src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp

```diff
@@ -21,6 +22,13 @@ void mark_shape_of_subgraphs::look_for_shape_of_subgraph(program_node& node) {
         return;
     }
 
+    // skip mark_node for broadcast node if dependency nodes are data and shape_of
+    auto& dependencies = node.get_dependencies();
```

Comment:
Can you please move this check to `mark_shape_of_subgraphs::can_mark_node()`?

Reply:
Addressed in commit [1fd12ac](https://github.com/openvinotoolkit/openvino/pull/27786/commits/1fd12aceb82c8fc7aada3dabfb139700e3a2931c).